gateway-js/src/index.ts

```diff
@@ -238,10 +239,23 @@ export class ApolloGateway implements GraphQLService {
     // 3. If config is `undefined`, use the default uplink URL
     if (isManagedConfig(this.config)) {
       const envEndpoint = process.env.APOLLO_SCHEMA_CONFIG_DELIVERY_ENDPOINT;
-      this.schemaConfigDeliveryEndpoint =
-        this.config.schemaConfigDeliveryEndpoint ??
-        envEndpoint ??
-        'https://uplink.api.apollographql.com/';
+      const endPoints = envEndpoint?.split(",") ?? null;
+      if (this.config.schemaConfigDeliveryEndpoint && !this.config.uplinkEndpoints) {
+        this.config.uplinkEndpoints = [this.config.schemaConfigDeliveryEndpoint];
```

Comment:
I think there's a bug here. What happens if I specify both `uplinkEndpoints` and `schemaConfigDeliveryEndpoint`? Presumably we'd want to take the newer option, but on this line we'd overwrite it and use it further down. I don't suspect we should ever overwrite the config which was provided to us.

Reply:
To clean up this bug and simplify the hierarchy / fallback logic a bit, we can use a chain off nullish coalesces like so, and I think it makes the the priority of the options fairly straightforward as well when reading the source.

```ts
const rawEndpointsString = process.env.APOLLO_SCHEMA_CONFIG_DELIVERY_ENDPOINT;
const envEndpoints = rawEndpointsString?.split(",") ?? null;

this.schemaConfigDeliveryEndpoints =
  this.config.uplinkEndpoints ??
  this.config.schemaConfigDeliveryEndpoint ??
  envEndpoints ?? [
    "https://uplink.api.apollographql.com/",
    "https://aws.uplink.api.apollographql.com/",
  ];
```

Reply:
I believe I have fixed this bug but not with the exact code you suggested because it mixes a string (`this.config.schemaConfigDeliveryEndpoint`) with arrays and the compiler does not like that.